### PR TITLE
Cudastream implementation for non-default cuda streams.

### PR DESCRIPTION
**Pull request by nickjbrowning**

### CODE REVIEW

File: sphericart-torch/CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@ check_language(CUDA)
 if(CMAKE_CUDA_COMPILER)
     enable_language(CUDA)
     set(CUDA_USE_STATIC_CUDA_RUNTIME OFF CACHE BOOL "" FORCE)
+    add_compile_definitions(CUDA_AVAILABLE)
```
Comment (Luthaf) [inline sphericart-torch/CMakeLists.txt:22]: this should be 
```suggestion
    target_compile_definitions(sphericart_torch PRIVATE CUDA_AVAILABLE)
```
or 
```suggestion
    target_compile_definitions(sphericart_torch PUBLIC CUDA_AVAILABLE)
```

`PRIVATE` if this is only set when compiling sphericart, `PUBLIC` if this is to be defined when using pre-compiled sphericart in another project.

**Review by Luthaf — COMMENTED** [review]
looks good overall, just a comment on CMake code!

File: sphericart-torch/CMakeLists.txt (hunk 1 above)
Comment (nickjbrowning) [reply sphericart-torch/CMakeLists.txt]: I've committed the PRIVATE statement.

**Review by Luthaf — APPROVED** [review]
I'll merge once CI is happy!

**Pull request merged**